### ISSUE TITLE
kernel-install: exit status is always 0

### ISSUE THREAD

**Issue by Nowa-Ammerlaan** [systemd/systemd#30087]
### systemd version the issue has been seen with

254.6

### Used distribution

Gentoo

### Linux kernel version used

6.6.1

### CPU architectures issue was seen on

x86_64

### Component

kernel-install

### Expected behaviour you didn't see

A failure in one of the plugin scripts should cause a failure of the whole kernel-install command, as also described in the manual:
```
EXIT STATUS
If every executable returns 0 or 77, 0 is returned, and a non-zero failure code otherwise.
```

### Unexpected behaviour you saw

A failure to install the kernel due to running out of disk space, results in a exit code of 0, indicating success. The issue was first reported downstream in Gentoo here: https://bugs.gentoo.org/917567

### Steps to reproduce the problem

- add `exit 1` at the top of any kernel-install plugin script
- run: `kernel-install add 6.6.1-gentoo-dist /usr/src/linux/arch/x86/boot/bzImage` or equivalent
- run: `echo $?` and find that `0` is returned

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
<snip>
dracut[I]: Stored kernel commandline:
dracut[I]:  root=UUID=2ae06136-d5d8-4901-b481-7df9b3fc2203 rootfstype=ext4 rootflags=rw,relatime
dracut[I]: *** Stripping files ***
dracut[I]: *** Stripping files done ***
dracut[I]: *** Creating image file '/tmp/kernel-install.staging.JgXLjg/uki.efi' ***
dracut[I]: Using auto-determined compression method 'gzip'
dracut[I]: Using UEFI kernel cmdline:
dracut[I]: root=PARTUUID=77ed6431-0d88-2846-9e21-835dc99f579b resume=PARTUUID=77ed6431-0d88-2846-9e21-835dc99f579b resume_offset=364099584 ro quiet splash apparmor=1 security=apparmor systemd.machine_id=204b505c963976da92752bb45b5adc9c drm.edid_firmware=DP-3:edid/hkc-tv-with-splitter-edid.bin
Enter engine key pass phrase:
Signing Unsigned original image
dracut[I]: *** Creating signed UEFI image file '/tmp/kernel-install.staging.JgXLjg/uki.efi' done ***
/usr/lib/kernel/install.d/90-uki-copy.install failed with exit status 1.

andrew-gentoo-pc linux # echo $?
0
```

**Comment by yuwata** [comment]
Thank you for the report. Fix is waiting in #30095.

**Comment by AndrewAmmerlaan** [comment]
Thanks for the quick fix!
